python/paddle/fluid/layers/nn.py

```diff
@@ -7860,8 +7860,18 @@ def relu(x, name=None):
     helper = LayerHelper('relu', **locals())
     dtype = helper.input_dtype(input_param_name='x')
     out = helper.create_variable_for_type_inference(dtype)
-    helper.append_op(
-        type="relu", inputs={"X": helper.input('x')}, outputs={"Out": out})
+    if x.type != core.VarDesc.VarType.LOD_TENSOR:
```

Comment:
I have two questions.

1. Why replace relu2 when LoDTensor

2. TODO in the future.
Is relu2 a good name as a new op? Maybe relu_with_mask (Or in the future, mask is an optional field for old relu)

Reply:
Replace with an automatic pass.